index.js

```diff
@@ -93,6 +101,14 @@ module.exports = function (log, indexesPath) {
     }
   })
 
+  log.deleted((offset) => {
+    const seq = getSeqFromOffset(offset)
+    if (seq < 0) return
+    updatePresentIndex(seq, offset, null)
+    // FIXME: debounce this save:
+    saveBitsetIndex('present', indexes['present'], () => {})
```

Comment:
Yes. Is this what you mean that there should be a WAL of deletes? :grimacing:

Reply:
I guess this is one of the side effects of "altering the past". SSB (and jitdb) so far has always moved forwards, but deletes are a way of altering the past, which complicates the code since it breaks a strong assumption we used to have.

Reply:
Yeah and compaction throws another wrench in the machine. I was looking at one of the tests that fails on the other branch to get a better idea of the problem and things that we thought were stable like [seq](https://github.com/ssbc/jitdb/commit/07514b739dcfb2f4370b61db1c673557e0a31537) are not. (btw isn't there still problem [here](https://github.com/ssbc/jitdb/blob/398d6f33f3d1cbbe3e4ef22d1cce35b67fdd9d43/operators.js#L514)?).

Reply:
Oh yeah, I haven't mentioned this, but I think my rewinding `seq - holesFound` is wrong, but it depends on fixing other things first (like #223) before we can trust seq. There's actually many ways how the rewinding is wrong.

E.g. 1: If `seq` is 3 but there were 1 deletes at seq 10, then you don't need to rewind, but we still do.

E.g. 2: if `seq` is 3 but there were 100 deletes at seq 200–300, then, we're going to rewind to seq -97, which is nonsensical.